src/main/room/main/G1HRPS/RoomManager.java

```diff
@@ -1,97 +1,305 @@
+	for(var i=0; i<room_list_.size();i++) {
+		if (room_list_.get(i).getRoomNumber()==roomNumber) {
+			room_list_.get(i).setStatus(status);
+		}
+	throw new UnsupportedOperationException();
+}
+}
+
+/**
+ * All room numbers in room_list_, organized by room type .
+ * e.g. Standard: Number: 2 out of 15 occupied 
+ * 				  Rooms: 02-21,02-22,02-33,02-44,02-55
+ * 		VIP:      Number: 3 out of 5 occupied
+ */
+public void GetRoomStatisticsByTypeOccupancyRate() {
+	int singleTotalCount;
+	int singleVacantCount;
+	String singleRoomNumber="";
+	int standardTotalCount;
+	int standardVacantCount;
+	String standardRoomNumber="";
+	int deluxeTotalCount;
+	int deluxeVacantCount;
+	String deluxeRoomNumber="";
+	int suiteTotalCount;
+	int suiteVacantCount;
+	String suiteRoomNumber="";
+	int VIPTotalCount;
+	int VIPVacantCount;
+	String VIPRoomNumber="";
+	
+	for(var i=0; i<room_list_.size();i++) {
+		RoomType type = room_list_.get(i).getRoomType();
+		switch(type) { //This switch block acts as a counter for different room types
+		
+		case Single:
+			singleTotalCount+=1;
+			singleRoomNumber=singleRoomNumber+room_list_.get(i).getRoomNumber()+",";
+			if(room_list_.get(i).getStatus()==RoomStatus.Vacant) {
+				singleVacantCount+=1;
+			}
+			break;	
+		case Standard:
+			standardTotalCount+=1;
+			if(room_list_.get(i).getStatus()==RoomStatus.Vacant) {
+				standardVacantCount+=1;
+				standardRoomNumber=singleRoomNumber+room_list_.get(i).getRoomNumber()+",";
+			}
+			break;
+		case Deluxe:
+			deluxeTotalCount+=1;
+			deluxeRoomNumber=deluxeRoomNumber+room_list_.get(i).getRoomNumber()+",";
+			if(room_list_.get(i).getStatus()==RoomStatus.Vacant) {
+				deluxeVacantCount+=1;
+			}
+			break;
+		case Suite:
+			suiteTotalCount+=1;
+			suiteRoomNumber=suiteRoomNumber+room_list_.get(i).getRoomNumber()+",";
+			if(room_list_.get(i).getStatus()==RoomStatus.Vacant) {
+				suiteVacantCount+=1;
+			}
+			break;
+		case VIP:
+			VIPTotalCount+=1;
+			VIPRoomNumber=VIPRoomNumber+room_list_.get(i).getRoomNumber()+",";
+			if(room_list_.get(i).getStatus()==RoomStatus.Vacant) {
+				VIPVacantCount+=1;
+			}
+			break;
+		default:
+			break;
 
-	/**
-	 * Occupancy rate of each room type.
-	 * e.g. Standard 2/5 occupied, VIP 1/5 occupied
-	 */
-	public void GetRoomStatisticsByTypeOccupancyRate() {
-		// TODO - implement RoomManager.GetRoomStatisticsByTypeOccupancyRate
-		throw new UnsupportedOperationException();
+			
+		}
+	singleRoomNumber =singleRoomNumber.substring(1,singleRoomNumber.length()-1); // removes the last comma in this String
+	standardRoomNumber =standardRoomNumber.substring(1,standardRoomNumber.length()-1);
+	suiteRoomNumber =suiteRoomNumber.substring(1,suiteRoomNumber.length()-1);
+	VIPRoomNumber =VIPRoomNumber.substring(1,VIPRoomNumber.length()-1);
+	deluxeRoomNumber =deluxeRoomNumber.substring(1,deluxeRoomNumber.length()-1);
+	System.out.println("Single: Number: "+ singleVacantCount +" out of "+singleTotalCount+" occupied");
```

Comment:
Also just to point out, you do `System.out.println("Single: Number: "+ singleVacantCount +" out of "+singleTotalCount+" occupied");` the variable is called vacant count, but at the end of the print you say it's occupied.